### PR TITLE
Configurable CA credential name for Client/Server TLS Settings

### PR DESCRIPTION
**Pull request by kfaseela**
Fixes https://github.com/istio/istio/issues/43966

### CODE REVIEW

File: pilot/pkg/features/pilot.go [hunk 1]
```diff
@@ -673,6 +673,9 @@ var (
 	// User should not rely on builtin resource labels, this flag will be removed in future releases(1.20).
 	EnableOTELBuiltinResourceLables = env.Register("ENABLE_OTEL_BUILTIN_RESOURCE_LABELS", false,
 		"If enabled, envoy will send builtin lables(e.g. node_name) via OTel sink.").Get()
+
+	// By default, Istio assumes CA credential name to have the format <credential>-cacert, use TLS_CUSTOM_CA_CREDENTIAL to override.
+	TLSCustomCaCredential = env.Register("TLS_CUSTOM_CA_CREDENTIAL", "", "The Custom CA credential name used in Client/Server TLS Settings").Get()
```
Comment (hzxuzhonghu) [inline pilot/pkg/features/pilot.go:678]: This is a hack way, and i think this cannot work:  how can you distinguish different  cacerts

Comment (howardjohn) [reply pilot/pkg/features/pilot.go:678]: https://github.com/kubernetes/enhancements/issues/3257 is probably the "proper" way to do that although its a bit early

**Comment by kfaseela** [comment]
As per offline discussion with @howardjohn this PR may not be acceptable because:

- mostly seems specific to 1 use case
- Having long term config options as hidden APIs in env vars isn't ideal and we are trying to get away from doing this
- Adds complexity to an already complexity scheme for picking the cert to use

The ideal way would be to add a new attribute to the API - ClientTLSSettings to support configuring caCredential, however we don't want to encourage more changes to APIs which will be deprecated in future. Hence closing this PR.

**Pull request closed without merging**